File: week6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Prompt the user to input a text
    text = get_string('Text: ')

    # Counting letters, words and sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculating the averages and the index
    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    # Rounding the index to the nearest integer
    grade = round(index)

    # Printing the result
    if grade < 1:
        print('Before Grade 1')
    elif grade > 15:
        print('Grade 16+')
    else:
        print(f'Grade {grade}')


# Function for counting letters
def count_letters(text):
    letters = 0
    length = len(text)
    for i in range(length):
        if text[i].isalpha():
            letters += 1
    return letters


# Function for counting words
def count_words(text):
    words = 1
    length = len(text)
    for i in range(length):
        if text[i] == ' ' and text[i + 1] != ' ':
            words += 1
    return words


# Function for counting sentences
def count_sentences(text):
    sentences = 0
    length = len(text)
    for i in range(length):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1
    return sentences


if __name__ == '__main__':
    main()

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

// Prototypes of the functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);


int main(void)
{
    // Prompt the user to input a text
    string text = get_string("Text: ");

    // Counting letters, words and sentences
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Calculating the averages and the index
    double L = 100.0 * letters / words;
    double S = 100.0 * sentences / words;
    double index = 0.0588 * L - 0.296 * S - 15.8;

    // Rounding the index to the nearest integer
    int grade = round(index);

    // Printing the result
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


// Function for counting letters
int count_letters(string text)
{
    int letters = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }

    return letters;
}


// Function for counting words
int count_words(string text)
{
    int words = 1;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            words++;
        }
    }

    return words;
}


// Function for counting sentences
int count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    return sentences;
}